frontend/src/routes/manage/shared.tsx

```diff
@@ -28,43 +27,50 @@ type ItemVars = {
         direction: SortDirection;
     };
     limit: number;
-    offset: number;
+    page: number;
 };
 
-type SharedProps = {
-    connection: EventConnection | SeriesConnection;
+type SharedProps<T> = {
+    connection: {
+        items: readonly T[];
+        totalCount: number;
+        pageInfo: {
+            hasNextPage: boolean;
+            hasPrevPage: boolean;
+        };
+    };
     vars: ItemVars;
-    additionalColumns?: ColumnProps[];
 };
 
-type ManageItemProps = SharedProps & {
+type SharedTableProps<T> = SharedProps<T> & {
+    RenderRow: (item: T) => ReactNode;
+    additionalColumns?: ColumnProps[];
+}
+
+type ManageItemProps<T extends Item> = SharedTableProps<T> & {
     titleKey: ParseKeys;
 }
 
 const LIMIT = 15;
 
-export const ManageItems: React.FC<ManageItemProps> = ({
+export const ManageItems = <T extends Item>({
     connection,
     vars,
     titleKey,
     additionalColumns,
-}) => {
+    RenderRow,
+}: ManageItemProps<T>) => {
     const { t } = useTranslation();
+    const router = useRouter();
 
-    const totalCount = connection.totalCount;
-    const limit = vars.limit ?? 15;
     const pageParam = new URLSearchParams(document.location.search).get("page");
-    const page = pageParam ? parseInt(pageParam, 10) : 1;
+    const page = pageParam ? parseInt(pageParam) : 1;
 
     useEffect(() => {
-        const maxPage = Math.max(Math.ceil(totalCount / limit), 1);
-
-        if (page > maxPage) {
-            window.location.href = `?page=${maxPage}`;
-        } else if (page < 1) {
-            window.location.href = "?page=1";
+        if (page < 1) {
+            router.replace(`${window.location.pathname}?page=1`);
         }
-    }, [page, totalCount, limit]);
+    }, [page]);
```

Comment:
It slightly bugs me that this component reads from the URL again. Feels like that should only be done in one place. But sure, that other place corrects invalid data to page 1, so there is no way for this component to see if there is actually garbage in the parameter. First I thought: maybe the redirect should happen in the route handler directly? But maybe even easier: do we really need that redirect? The only way to get an invalid value in the `page` parameter is that a user manually edits the URL. Then we default to page 1. Without the redirect, the garbage remains in the URL while Tobira shows page 1. But is that really a big deal? All links in Tobira are correct and as soon as the user clicks anything, the URL is fine again. So I guess I would simply remove this?